src/nrncvode/netcvode.cpp

```diff
@@ -5942,6 +5942,18 @@ void NetCvode::deliver_net_events(NrnThread* nt) { // for default method
     // but I do not want to affect the case of not using a bin queue.
 
 	if (nrn_use_bin_queue_) {
+		// it was noticed on binq + compressed spike exchange +
+		// threads that a transferred event may be languishing in
+		// the interthread event buffer. Perhaps this is better done
+		// as a multithread job at the end of nrn_spike_exchange
+		// instead of every time step --- but here we are
+		// already in a multithread job, so what is the overhead of
+		// starting such a small one in nrn_spike_exchange.
+		extern bool nrn_use_compress_;
+		if (nrn_use_compress_ && nrn_nthread > 1) {
+			p[tid].enqueue(this, nt);
+		}
+
```

Comment:
Perhaps all of this should done in a similar way to that chosen for coreneuron. I.e. In coreneuron it was put at the end of ```nrnmpi_spike_exchange_compressed_impl``` and in ```BBS_netpar_solve```